src/components/field-label/field-label.tsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import invariant from 'tiny-invariant';
+import IconButton from '../buttons/icon-button';
+import { InformationIcon } from '../icons';
+import Text from '../typography/text';
+import Label from '../label';
+import Constraints from '../constraints';
+import Spacings from '../spacings';
+import styles from './field-label.mod.css';
+
+type HorizontalConstraintOfFieldLabel = 's' | 'm' | 'l' | 'xl' | 'scale';
+type FieldLabelProps = {
+  title: React.ReactNode;
+  hint: React.ReactNode;
+  description: React.ReactNode;
+  onInfoButtonClick?: () => void;
+  hintIcon: React.ReactNode;
+  badge: React.ReactNode;
+  hasRequiredIndicator?: boolean;
+  htmlFor?: string;
+  horizontalConstraint?: HorizontalConstraintOfFieldLabel;
+};
+
+export const FieldLabel: React.FC<FieldLabelProps> = props => {
+  invariant(
+    props.hint && !props.hintIcon,
+    'ui-kit/FieldLabel: please pass `hintIcon` when label has hint.'
+  );
```

Comment:
This is the same as with an input. I can't find a way to have a type "reflect" on itself.